tests/validators/test_float.py

```diff
@@ -174,3 +178,106 @@ def test_float_key(py_and_json: PyAndJson):
     assert v.validate_test({'1.5': 1, '2.4': 2}) == {1.5: 1, 2.4: 2}
     with pytest.raises(ValidationError, match='Input should be a valid number'):
         v.validate_test({'1.5': 1, '2.5': 2}, strict=True)
+
+
+@pytest.mark.parametrize(
+    'input_value,only_finite,expected',
+    [
+        ('NaN', False, lambda x: math.isnan(x)),
+        (
+            'NaN',
+            True,
+            Err(
+                'Input should be finite (neither infinite nor NaN) '
+                "[kind=float_finite, input_value='NaN', input_type=str]"
+            ),
+        ),
+        ('+inf', False, lambda x: math.isinf(x) and x > 0),
+        (
+            '+inf',
+            True,
+            Err(
+                'Input should be finite (neither infinite nor NaN) '
+                "[kind=float_finite, input_value='+inf', input_type=str]"
+            ),
+        ),
+        ('+infinity', False, lambda x: math.isinf(x) and x > 0),
+        (
+            '+infinity',
+            True,
+            Err(
+                'Input should be finite (neither infinite nor NaN) '
+                "[kind=float_finite, input_value='+infinity', input_type=str]"
+            ),
+        ),
+        ('-inf', False, lambda x: math.isinf(x) and x < 0),
+        (
+            '-inf',
+            True,
+            Err(
+                'Input should be finite (neither infinite nor NaN) '
+                "[kind=float_finite, input_value='-inf', input_type=str]"
+            ),
+        ),
+        ('-infinity', False, lambda x: math.isinf(x) and x < 0),
+        (
+            '-infinity',
+            True,
+            Err(
+                'Input should be finite (neither infinite nor NaN) '
+                "[kind=float_finite, input_value='-infinity', input_type=str]"
+            ),
+        ),
+        ('0.7', False, lambda x: x == 0.7),
+        ('0.7', True, lambda x: x == 0.7),
+        (
+            'pika',
+            False,
+            Err(
+                'Input should be a valid number, unable to parse string as an number '
+                "[kind=float_parsing, input_value='pika', input_type=str]"
+            ),
+        ),
+        (
+            'pika',
+            True,
+            Err(
+                'Input should be a valid number, unable to parse string as an number '
+                "[kind=float_parsing, input_value='pika', input_type=str]"
+            ),
+        ),
+    ],
+)
+def test_non_finite_json_values(py_and_json: PyAndJson, input_value, only_finite, expected):
```

Comment:
The test is a bit verbose but more explicit and readable than with more parametrization. Do not hesitate to ask for a change or do it yourself @samuelcolvin.